File: apps/desktop/src/main/platform.ts
/**
 * 判断是否为 Mac 系统
 * @returns 是否为 Mac 系统
 */
export function isMac() {
    return process.platform === "darwin";
}

/**
 * 判断是否为 Windows 系统
 * @returns 是否为 Windows 系统
 */
export function isWin() {
    return process.platform === "win32";
}

/**
 * 判断是否为 Linux 系统
 * @returns 是否为 Linux 系统
 */
export function isLinux() {
    return process.platform === "linux";
}

File: apps/photasa/src-tauri/src/commands/platform.rs
/*!
 * 平台检测命令 (RFC 0091)
 * 返回与 Node process.platform 一致的值，供前端 isMac 等使用
 */

/// 返回当前操作系统标识："darwin" | "win32" | "linux"
#[tauri::command]
pub fn get_platform() -> String {
    match std::env::consts::OS {
        "macos" => "darwin".to_string(),
        "windows" => "win32".to_string(),
        "linux" => "linux".to_string(),
        other => other.to_string(),
    }
}

/// 应用版本（与 Cargo 包版本一致，供 legacy-api getAppVersion）
#[tauri::command]
pub fn get_app_version() -> String {
    env!("CARGO_PKG_VERSION").to_string()
}
